tests/unit/test_fracture_network.py

```diff
@@ -314,6 +313,79 @@ def test_add_networks_preserve_tags(self):
 
+    def test_triangle_domain_no_fracs(self):
+        line_1 = np.array([[0, 1], [0, 0]])
+        line_2 = np.array([[1, 0], [0, 1]])
+        line_3 = np.array([[0, 0], [1, 0]])
+        lines = [line_1, line_2, line_3]
+        domain = pp.Domain(polytope=lines)
+
+        mesh_size = 0.5
+        mesh_kwargs = {"mesh_size_frac": mesh_size}
+        mdg = pp.create_fracture_network(domain=domain).mesh(mesh_kwargs)
+        sd = mdg.subdomains(dim=2)[0]
+
+        self.assertTrue(np.isclose(sd.num_cells, 7))
+        self.assertTrue(np.isclose(sd.num_faces, 14))
+        self.assertTrue(np.isclose(sd.num_nodes, 8))
+
+    def test_triangle_domain_with_fracs(self):
+        line_1 = np.array([[0, 1], [0, 0]])
+        line_2 = np.array([[1, 0], [0, 1]])
+        line_3 = np.array([[0, 0], [1, 0]])
+        lines = [line_1, line_2, line_3]
+        domain = pp.Domain(polytope=lines)
+
+        pts = np.array([[0, 1], [0, 1]])
+        edges = np.array([[0], [1]])
+        fracs = pts_edges_to_linefractures(pts, edges)
+
+        mesh_size = 0.5
+        mesh_kwargs = {"mesh_size_frac": mesh_size}
+        mdg = pp.create_fracture_network(fracs, domain=domain).mesh(mesh_kwargs)
+
+        sd_2d = mdg.subdomains(dim=2)[0]
+        self.assertTrue(np.isclose(sd_2d.num_cells, 12))
+        self.assertTrue(np.isclose(sd_2d.num_faces, 24))
+        self.assertTrue(np.isclose(sd_2d.num_nodes, 14))
+
+        sd_1d = mdg.subdomains(dim=1)[0]
+        self.assertTrue(np.isclose(sd_1d.num_cells, 2))
+        self.assertTrue(np.isclose(sd_1d.num_faces, 3))
+        self.assertTrue(np.isclose(sd_1d.num_nodes, 3))
+
+    def test_pentagon_domain_with_fracs(self):
+        line_1 = np.array([[0, 1], [0, 0]])
+        line_2 = np.array([[1, 1], [0, 1]])
+        line_3 = np.array([[1, 0.5], [1, 0.5]])
+        line_4 = np.array([[0.5, 0], [0.5, 1]])
+        line_5 = np.array([[0, 0], [1, 0]])
+        lines = [line_1, line_2, line_3, line_4, line_5]
+        domain = pp.Domain(polytope=lines)
+
+        pts = np.array([[0.2, 0.8], [0.6, 0.6]])
+        edges = np.array([[0], [1]])
+        fracs = pts_edges_to_linefractures(pts, edges)
+
+        mesh_size = 0.5
+        mesh_kwargs = {"mesh_size_frac": mesh_size}
+        mdg = pp.create_fracture_network(fracs, domain=domain).mesh(mesh_kwargs)
+
+        sd_2d = mdg.subdomains(dim=2)[0]
+        self.assertTrue(np.isclose(sd_2d.num_cells, 72))
+        self.assertTrue(np.isclose(sd_2d.num_faces, 123))
+        self.assertTrue(np.isclose(sd_2d.num_nodes, 52))
+
+        sd_1d = mdg.subdomains(dim=1)[0]
+        self.assertTrue(np.isclose(sd_1d.num_cells, 2))
+        self.assertTrue(np.isclose(sd_1d.num_faces, 3))
+        self.assertTrue(np.isclose(sd_1d.num_nodes, 3))
+
+        sd_1d = mdg.subdomains(dim=1)[1]
+        self.assertTrue(np.isclose(sd_1d.num_cells, 2))
+        self.assertTrue(np.isclose(sd_1d.num_faces, 3))
+        self.assertTrue(np.isclose(sd_1d.num_nodes, 3))
```

Comment:
Sorry, but this test will not work. The number of cells is not invariant if Gmsh changes its algorithms. Also, does such a acceptance criterion really pinpoint the part of the code you want to test?

Reply:
Tests are now improved (I hope) according to @jhabriel suggestions